xwiki-platform-core/xwiki-platform-office/xwiki-platform-office-importer/src/main/java/org/xwiki/officeimporter/converter/OfficeConverterResult.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.officeimporter.converter;
+
+import java.io.File;
+import java.util.Set;
+
+import org.xwiki.stability.Unstable;
+
+/**
+ * Provide the result of an office file conversion.
+ * This interface mainly returns the paths of the various files created during the conversion, but it also allows to
+ * cleanup all files created.
+ *
+ * @version $Id$
+ * @since 13.1RC1
+ */
+@Unstable
+public interface OfficeConverterResult
+{
+    /**
+     * @return the output file of the conversion.
+     */
+    File getOutputFile();
+
+    /**
+     * @return the directory containing all output files.
+     */
+    File getOutputDirectory();
+
+    /**
+     * @return all files created during the conversion, including {@link #getOutputFile()}.
+     */
+    Set<File> getAllFiles();
```

Comment:
Maybe we should expose Path instead of File in new APIs. It's not like File was deprecated but using NIO APIs is kind of the recommended way now.

Reply:
> Maybe we should expose Path instead of File in new APIs

I actually started the implem with File and I finally reverted it to use Path because all our usages actually need File, so it sounded easier to directly use File than calling everywhere `toFile`.